Sources/Event.swift

```diff
@@ -861,6 +861,63 @@ extension Signal.Event {
 			}
 		}
 	}
+	
+	internal static func collect(every interval: DispatchTimeInterval, on scheduler: DateScheduler, skipEmpty: Bool, discardsWhenCompleted: Bool) -> Transformation<[Value], Error> {
+		return { action, lifetime in
+			let values = Atomic<CollectEveryState<Value>>(.init(skipEmpty: skipEmpty))
+			let d = SerialDisposable()
+			
+			d.inner = scheduler.schedule(after: scheduler.currentDate.addingTimeInterval(interval), interval: interval, leeway: interval * 0.1) {
+				let (currentValues, isCompleted) = values.modify { ($0.collect(), $0.isCompleted) }
+				if let currentValues = currentValues {
+					action(.value(currentValues))
+				}
+				if isCompleted {
+					action(.completed)
```

Comment:
@mdiep Should I call `d.dispose()` manually here?

Reply:
Yes, I think so. Otherwise the timer would continue firing.

Reply:
`d` is disposed already as part of the lifetime observer, so repeating it here is not necessary.